### PR TITLE
test: improve ESLint configuration

### PR DESCRIPTION
**Pull request by tancnle**
# Description

We rarely apply ESLint configuration to all files. Hence, we extract rules specific to file types to have a stricker separation of concerns.
    
- Simplify recommended rules
- Exclude folders from ESLint 
- Remove duplicated rules (`no-unused-vars`) reported by both ESLint and TypeScript ESLint 

## Type of change

- [x] Refactoring linting configuration

# How Has This Been Tested?

1. Check out this branch
2. Run `npm run lint`
3. Observe the log showing a reduced set of problems, no errors

```shell
# Before
✖ 1841 problems (0 errors, 1841 warnings)
  0 errors and 365 warnings potentially fixable with the `--fix` option.

# After
✖ 1549 problems (0 errors, 1549 warnings)
  0 errors and 365 warnings potentially fixable with the `--fix` option.
```

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,61 +1,60 @@
...
+  "parser": "@typescript-eslint/parser",
+  "parserOptions": {
+    "project": ["tsconfig.*?.json"],
+    "createDefaultProgram": true
   },
-  "extends": ["eslint:recommended", "plugin:prettier/recommended"],
+  "plugins": ["@typescript-eslint"],
+  "extends": [
+    "plugin:@angular-eslint/recommended",
+    "plugin:@typescript-eslint/recommended",
```
Comment (tancnle) [inline .eslintrc.json:11]: This includes rules from `eslint:recommended` (see [note](https://typescript-eslint.io/docs/linting/configs/#recommended)).

File: .eslintrc.json [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
+      "parser": "@typescript-eslint/parser",
       "parserOptions": {
         "project": ["tsconfig.*?.json"],
         "createDefaultProgram": true
       },
-      "extends": ["plugin:@typescript-eslint/recommended", "plugin:@angular-eslint/recommended"],
+      "plugins": ["@typescript-eslint"],
+      "extends": [
+        "plugin:@angular-eslint/recommended",
+        "plugin:@typescript-eslint/recommended",
```
Comment (tancnle) [inline .eslintrc.json:14]: This includes rules from `eslint:recommended` (see [note](https://typescript-eslint.io/docs/linting/configs/#recommended)) hence we can remove it.

File: .eslintrc.json [hunk 3]
```diff
@@ -43,7 +43,7 @@
         "@typescript-eslint/no-this-alias": "warn",
         "no-dupe-class-members": "warn",
         "no-prototype-builtins": "warn",
-        "no-unused-vars": "warn",
+        "no-unused-vars": "off",
```
Comment (tancnle) [inline .eslintrc.json:46]: This issue show up twice in the log as it is picked up by ESLint and TypeScript ESLint

```bash
npm run lint
<snip>
src/app/units/states/tasks/inbox/directives/staff-task-list/staff-task-list.component.ts        
   14:26   warning  'Unit' is defined but never used                no-unused-vars                                                               
   14:26   warning  'Unit' is defined but never used                @typescript-eslint/no-unused-vars     
<snip>
```

File: .eslintrc.json [hunk 4]
```diff
@@ -53,19 +53,21 @@
...
+      "extends": ["plugin:@angular-eslint/template/recommended", "plugin:prettier/recommended"],
       "rules": {
         "max-len": [
           "warn",
           {
             "code": 140
           }
-        ]
+        ],
+        "prettier/prettier": "warn"
```
Comment (tancnle) [inline .eslintrc.json:70]: Prettier is capable of formatting HTML as well.

**Review by PerryRose — APPROVED** [review]
Great job Tan

**Review by RayGuo-ergou — APPROVED** [review]
LGTM. thanks for the explainations.

**Comment by tancnle** [comment]
Thank you for reviewing @PerryRose and @RayGuo-ergou. As always, really appreciated! 🙏🏼

**Pull request merged**